sstables/compaction_manager.cc

```diff
@@ -596,50 +596,63 @@ future<> compaction_manager::rewrite_sstables(column_family* cf, sstables::compa
     task->compacting_cf = cf;
     task->type = options.type();
     _tasks.push_back(task);
-    _stats.pending_tasks++;
 
-    task->compaction_done = repeat([this, task, options, get_func = std::move(get_func)] () mutable {
+    auto sstables = std::make_unique<std::vector<sstables::shared_sstable>>(get_func(*cf));
+    auto w = sstables.get();
+    _stats.pending_tasks += sstables->size();
+
+    task->compaction_done = do_until([w] { return !w->empty(); }, [this, task, options, w] () mutable {
```

Comment:
do_until's interface is sometime counter-intuitive because its first parameter is stop condition, not the "keep going" condition, which is what we're most familiar with in programming, ex: while(condition) { do }
do_until([w] { return !w->empty(); } must be do_until([w] { return w->empty(); })

Reply:
I go on saying we should perhaps deprecate do_until(stop_condition, action) in favor of a new do_while(condition, action). I prefer the later much more.

Reply:
@glommer That's probably the reason behind the test failure in debug mode, right?

Reply:
Maybe, thanks a lot Raphael! An update : I was probably compiling and running different binaries because Raphael and I found out that indeed the test was not failing before. Thinking it did thew me off track

(that's probably an artifact of the `test_g` vs `test` thing)

Updating